src/limestone-query.ts

```diff
@@ -0,0 +1,246 @@
+    return new LimestoneQueryForSingleSymbol({
+      ...this.params,
+      startDate: date,
+    });
+  }
+
+  /**
+   * Configures query to fetch the price in a time range. toDate method should go after the fromDate
+   * @param date - End date/time for the time range
+   *
+   * @returns query object
+   */
+  toDate(date: ConvertableToDate): LimestoneQueryExecutable<PriceData[]> {
+    if (this.params.startDate === undefined) {
+      throw new Error("Please specify fromDate before using toDate");
+    }
+    return this.getExecutableQuery<PriceData[]>({ endDate: date });
+  }
+
+  /**
+   * Configures query to fetch the price for the last few hours
+   * @param hoursCount - Number of hours in the time range
+   *
+   * @returns query object
+   */
+  forLastHours(hoursCount: number): LimestoneQueryExecutable<PriceData[]> {
+    const endDate = Date.now();
+    return this.getExecutableQuery({
+      endDate,
+      startDate: endDate - hoursCount * 3600 * 1000,
+      interval: 600 * 1000,
+    });
+  }
+
+  /**
+   * Configures query to fetch the price for the last few days
+   * @param daysCount - Number of days in the time range
+   *
+   * @returns query object
+   */
+  forLastDays(daysCount: number): LimestoneQueryExecutable<PriceData[]> {
+    const endDate = Date.now();
+    return this.getExecutableQuery({
+      endDate,
+      startDate: endDate - daysCount * 24 * 3600 * 1000,
+      interval: 3600 * 1000,
+    });
+  }
+}
+
+export class LimestoneQueryForSeveralSymbols extends LimestoneQueryForSingleOrSeveralSymbols<{
+  [symbol: string]: PriceData;
+}> {
+  constructor(params: QueryParams) {
+    super(params);
+  }
+}
+
+export class LimestoneQueryExecutable<QueryResultType> {
+  private params: QueryParams;
+
+  constructor(params = {}) {
+    this.params = {
+      symbols: [],
+      ...params,
+    };
+  }
+
+  /**
+   * Executes the query
+   *
+   * @returns Promise resolving the query result (result type depends on the query)
+   */
+  async exec(): Promise<QueryResultType> {
+    const limestone = new LimestoneApi();
+    const symbols = this.params.symbols;
+    if (symbols.length > 0) {
+      const symbolOrSymbols = symbols.length === 1 ? symbols[0] : symbols;
+      const { startDate, endDate, date, interval } = this.params;
+
+      if ([startDate, endDate, date].every((el) => el === undefined)) {
+        // Fetch the latest price
+        return (await limestone.getPrice(
+          symbolOrSymbols as any,
+          this.params as any,
+        )) as any;
+      } else {
+        // Fetch the historical price
+        if (
+          startDate !== undefined &&
+          endDate !== undefined &&
+          interval === undefined
+        ) {
+          const diff = getTimeDiff(startDate, endDate);
+          if (diff >= 24 * 3600 * 1000) {
+            this.params.interval = 3600 * 1000;
+          } else {
+            this.params.interval = 1;
+          }
+        }
+
+        // TODO: check types
```

Comment:
Let's put a task in the backlog to improve types handling in fluent interface.

Reply:
Added